File: webapp/src/models/MusicGenre.ts
export const MusicGenre = {
    PROGRESSIVE_ROCK: 'PROGRESSIVE_ROCK',
    HIP_HOP: 'HIP_HOP',
    PUNK_ROCK: 'PUNK_ROCK'
} as const;

export type MusicGenre = typeof MusicGenre[keyof typeof MusicGenre];

File: labs/src/main/java/com/evelina/labs/models/MusicGenre.java
package com.evelina.labs.models;

import jakarta.xml.bind.annotation.XmlEnum;
import jakarta.xml.bind.annotation.XmlEnumValue;

@XmlEnum (String.class)
public enum MusicGenre {
    @XmlEnumValue("PROGRESSIVE_ROCK")
    PROGRESSIVE_ROCK,
    @XmlEnumValue("HIP_HOP")
    HIP_HOP,
    @XmlEnumValue("PUNK_ROCK")
    PUNK_ROCK;
}